### PR TITLE
feat(openapi-generator): adds typescript-angular template

### PR DESCRIPTION
**Pull request by mario-subo**

### CODE REVIEW

File: src/openapiGenerator/templates/typescript-angular/api.service.mustache [hunk 1]
```diff
@@ -0,0 +1,546 @@
+{{>licenseInfo}}
+/* tslint:disable:no-unused-variable member-ordering */
+
+import { Inject, Injectable, Optional }                      from '@angular/core';
+{{#useHttpClient}}
+import { HttpClient, HttpHeaders, HttpParams,
+         HttpResponse, HttpEvent, HttpParameterCodec }       from '@angular/common/http';
+import { CustomHttpParameterCodec }                          from '../encoder';
```
Comment (DanSallau) [inline src/openapiGenerator/templates/typescript-angular/api.service.mustache:8]: The imports here looks like project specific , or is it something common to all angular projects ? If its common to angular projects, at the very least can we typescript us modular importation instead like 
`import { CustomHttpParameterCodec }                          from '@utils/encoder';`

File: src/openapiGenerator/options/customOptions.ts [hunk 2]
```diff
@@ -76,6 +76,10 @@ const templates: Array<SEBTemplate> = [
     {
         generator: "typescript-axios",
         templatePath: "./node_modules/@sebgroup/frontend-tools/dist/openapiGenerator/templates/typescript-axios"
+    },
+    {
+        generator: "typescript-angular",
+        templatePath: "./node_modules/@sebgroup/frontend-tools/dist/openapiGenerator/templates/typescript-angular"
```
Comment (DanSallau) [inline src/openapiGenerator/options/customOptions.ts:82]: So `typescript-angular` here uses `HttpClientModule` ? Just wondering.

File: src/openapiGenerator/templates/typescript-angular/api.service.mustache (hunk 1 above)
Comment (mario-subo) [reply src/openapiGenerator/templates/typescript-angular/api.service.mustache:8]: that part is unchanged from the openapi mustache file https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L7

It points to this file which also gets generated and is reused in all api services 
https://github.com/OpenAPITools/openapi-generator/blob/ec39005ae528f88522caedd91db91c0b59624d94/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/encoder.ts

File: src/openapiGenerator/options/customOptions.ts (hunk 2 above)
Comment (mario-subo) [reply src/openapiGenerator/options/customOptions.ts:82]: yep 

see this example of a generated file https://github.com/OpenAPITools/openapi-generator/blob/ec39005ae528f88522caedd91db91c0b59624d94/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/api/pet.service.ts#L36

**Pull request merged**